runhouse/rns/hardware/cluster.py

```diff
@@ -314,32 +315,38 @@ def check_grpc(self, restart_grpc_server=True):
 
         if not self.client:
             try:
-                self.connect_grpc()
+                self.connect_server_client()
+                cluster_config = self.config_for_rns
+                if "sky_state" in cluster_config.keys():
+                    # a bunch of setup commands that mess up json dump
+                    del cluster_config["sky_state"]
+                self.client.check_server(cluster_config=cluster_config)
             except (
-                grpc.RpcError,
+                requests.exceptions.ConnectionError,
                 sshtunnel.BaseSSHTunnelForwarderError,
             ):
                 # It's possible that the cluster went down while we were trying to install packages.
                 if not self.is_up():
                     self.up_if_not()
                 else:
-                    self.restart_grpc_server(resync_rh=False)
+                    self.restart_server(resync_rh=False)
+        return
 
         if self.is_connected():
             return
 
-        self.connect_grpc()
+        self.connect_server_client()
         if self.is_connected():
             return
 
-        if restart_grpc_server:
-            self.restart_grpc_server(resync_rh=False)
-            self.connect_grpc()
+        if restart_server:
+            self.restart_server(resync_rh=False)
+            self.connect_server_client()
             if self.is_connected():
                 return
 
-            self.restart_grpc_server(resync_rh=True)
-            self.connect_grpc()
+            self.restart_server(resync_rh=True)
```

Comment:
why do we try again here? is this to see if tunnel is already up?

Reply:
This was legacy, fixed!